port/linuxriscv64/omrsignal_context.c

```diff
@@ -89,75 +75,195 @@ infoForSignal(struct OMRPortLibrary *portLibrary, struct OMRUnixSignalInfo *info
 		*name = "Handler2";
 		*value = &info->handlerAddress2;
 		return OMRPORT_SIG_VALUE_ADDRESS;
-	
+
 	case OMRPORT_SIG_SIGNAL_INACCESSIBLE_ADDRESS:
 	case 6:
 		/* si_code > 0 indicates that the signal was generated by the kernel */
-		if ( NULL != info->sigInfo && 0 < info->sigInfo->si_code ) {
+		if (info->sigInfo->si_code > 0) {
 			if ((info->sigInfo->si_signo == SIGBUS) || (info->sigInfo->si_signo == SIGSEGV)) {
 				*name = "InaccessibleAddress";
 				*value = &info->sigInfo->si_addr;
 				return OMRPORT_SIG_VALUE_ADDRESS;
 			}
 		}
-		break;
-
-	// case OMRPORT_SIG_SIGNAL_ADDRESS:
-	// case 7:
-	// 	*name = "fault_address";
-	// 	*value = &((struct sigcontext *)&info->platformSignalInfo.context->uc_mcontext)->fault_address;
-	// 	return OMRPORT_SIG_VALUE_ADDRESS;
-
+		return OMRPORT_SIG_VALUE_UNDEFINED;
 
 	default:
-		break;
+		return OMRPORT_SIG_VALUE_UNDEFINED;
 	}
-	
-	return OMRPORT_SIG_VALUE_UNDEFINED;
 }
 
+uint32_t
+infoForFPR(struct OMRPortLibrary *portLibrary, OMRUnixSignalInfo *info, int32_t index, const char **name, void **value)
+{
+	struct sigcontext *const context = (struct sigcontext *)&info->platformSignalInfo.context->uc_mcontext;
+	const char *n_fpr[NFPRS] = {
+			"FT0",
+			"FT1",
+			"FT2",
+			"FT3",
+			"FT4",
+			"FT5",
+			"FT6",
+			"FT7",
+			"FS0",
+			"FS1",
+			"FA0",
+			"FA1",
+			"FA2",
+			"FA3",
+			"FA4",
+			"FA5",
+			"FA6",
+			"FA7",
+			"FS2",
+			"FS3",
+			"FS4",
+			"FS5",
+			"FS6",
+			"FS7",
+			"FS8",
+			"FS9",
+			"FS10",
+			"FS11",
+			"FT8",
+			"FT9",
+			"FT10",
+			"FT11",
+			"FCSR"
+	};
 
-/**
- * Walk the 4k reserve space to get the floating point registers since the layout differ 
- * per-chip
- * @param	magic_to_find	input	magic number #define in the linux kernel (sigcontext.h)\
- * that defines which struct the reserved area is at a given space
- * @param	reserved_space	input	array of bytes that holds all the struct
- * @see linux/arch/riscv/include/uapi/asm/sigcontext.h for the defines and information to why we need
- * to walk the space
+	*name = "";
 
- */
-void*
-walkReserveSpace(uint32_t magic_to_find, uint8_t *reserved_space)
-{
-	Assert_PRT_true(0 && "Not yet implemented");
-	return NULL;
+	if ((index >= 0) && (index < NFPRS)) {
+		*name = n_fpr[index];
+		*value = &(context->fpregs[index]);
+		return OMRPORT_SIG_VALUE_FLOAT_64;
+	} else {
```

Comment:
nit pick - the `else {` isn't required as the `if` block returns.

Clearer to write:
```c
if ((index >= 0) && (index < NFPRS)) {
	.....
	.....
	return OMRPORT_SIG_VALUE_FLOAT_64;
}
return OMRPORT_SIG_VALUE_UNDEFINED;
```
as it becomes obvious there is no other paths to the end of the method

Reply:
Updated the code as suggested above.